### ISSUE TITLE
Remove the `error_handler` field

### ISSUE THREAD

**Issue by mokurin000** [Deaths-Door/musixmatch-rs#4]
In current API, we will send error to the handler, and return an `Option<T>`, which just make users harder to get Error

**Comment by Deaths-Door** [comment]
Sorry for the late reply, I've been busy with exams. Since I don’t have time to work on this at the moment, feel free to give it a try yourself! I initially thought it would be nicer to have centralized error handling to keep the implementation clean and consistent, but I'm open to your approach if it works better for this case. Let me know how it goes!

**Comment by mokurin000** [comment]
> Sorry for the late reply,

That's OK

centralized error handling might be a good idea for applications, but it's not idiom to rust, By the way, how could a user collect errors for parallel calls, if he just sends errors to a channel?

**Comment by Deaths-Door** [comment]
> Centralized error handling might be a good idea for applications.

Yeah, I originally wrote the library for a specific application, so that checks out.

> How could a user collect errors for parallel calls if they just send errors to a channel?

If you’re asking how to actually implement this (and if not correct me)

one way is to use a channel (like std::sync::mpsc) to collect errors and aggregate them later. With the latest released API (not the unfinished one I’m working on), you could use the error_resolver to centralize error collection. For example:
```rust
#[tokio::main]
async fn main() {
    // Create a channel for collecting errors
    let (tx, rx) = mpsc::channel();

    // Create the client with an error resolver that sends errors to the channel
    let client = MusixAbgleich::new("...",Box::new(move |err| {
            tx.send(err).unwrap();
        }));

    let mut handles = Vec::new();
    for _ in 0..5 {
        let client = &client; // Reference to the client
        handles.push(task::spawn(async move {
           // Fetch something here
        }));
    }

    // Wait for all tasks to complete
    for handle in handles {
      // ignoring any errors here
        handle.await.unwrap();
    }

    // Collect all errors from the channel
    let errors: Vec<_> = rx.try_iter().collect();
    println!("Collected errors: {:?}", errors);
}
```

**Comment by mokurin000** [comment]
> > Centralized error handling might be a good idea for applications.
> 
> Yeah, I originally wrote the library for a specific application, so that checks out.
> 
> > How could a user collect errors for parallel calls if they just send errors to a channel?
> 
> If you’re asking how to actually implement this (and if not correct me)
> 
> one way is to use a channel (like std::sync::mpsc) to collect errors and aggregate them later. With the latest released API (not the unfinished one I’m working on), you could use the error_resolver to centralize error collection. For example:
> 
> ```rust
> #[tokio::main]
> async fn main() {
>     // Create a channel for collecting errors
>     let (tx, rx) = mpsc::channel();
> 
>     // Create the client with an error resolver that sends errors to the channel
>     let client = MusixAbgleich::new("...",Box::new(move |err| {
>             tx.send(err).unwrap();
>         }));
> 
>     let mut handles = Vec::new();
>     for _ in 0..5 {
>         let client = &client; // Reference to the client
>         handles.push(task::spawn(async move {
>            // Fetch something here
>         }));
>     }
> 
>     // Wait for all tasks to complete
>     for handle in handles {
>       // ignoring any errors here
>         handle.await.unwrap();
>     }
> 
>     // Collect all errors from the channel
>     let errors: Vec<_> = rx.try_iter().collect();
>     println!("Collected errors: {:?}", errors);
> }
> ```

Yeah, a channel would work, if the user don't care where the error was from:

For example, I spawned two different tasks to run (For example by JoinSet), and both of them failed

Without manual task-id management, How could I detect which error belongs a task?

**Comment by Deaths-Door** [comment]
Of course, in this case, it would be better to return the error. **IF** it’s still necessary,  the current API could be retained (though personally, I’m against it based on your argument).

That said, since I don’t have the time right now due to my exams, feel free to start on it if you’d like. I’d guess the best approach would be to build on [this PR](https://github.com/Deaths-Door/musixmatch-rs/pull/3). In that PR, I was working on making the API more type-safe and easier to use across calls. For example, instead of using `page_size` as a plain integer, I was experimenting with a newtype to make it more flexible and better suited for passing around. That was the idea, anyway.

Just as a side note, I noticed (though I haven’t fully tested it across all the endpoints) that deserialization seems to be failing on successful responses. That’s something to look into
